src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -288,11 +288,14 @@ def _deploy_arm_template_core(cli_ctx, resource_group_name,
                                       parameters=parameters, mode=mode, on_error_deployment=on_error_deployment)
 
     smc = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES, aux_subscriptions=aux_subscriptions)
+
+    validation_result = smc.deployments.validate(resource_group_name=resource_group_name, deployment_name=deployment_name, properties=properties)
+
+    if validation_result and validation_result.error:
+        raise CLIError(validation_result.error)
```

Comment:
It seems you do not check validate_only in this method now, is there any reason for this? And why do you not use sdk_no_wait?

Reply:
we want to do validate for both validate and create, so we are not checking validate_only. we don't need sdk_no_wait because the validate API is not a long running operation.